### PR TITLE
DAOS-623 ci: Update the existing (Jenkins) commit statuses

### PR DESCRIPTION
**Pull request by brianjmurrell**
From the corresponding GHA workflows.

### CODE REVIEW

File: .github/workflows/rpm-build-and-test.yml [hunk 1]
```diff
@@ -297,6 +303,12 @@ jobs:
         with:
           name: ${{ env.DISTRO_NAME }} ${{ env.DISTRO_VERSION }} RPM repository
           path: ${{ env.REPO_PATH}}${{ github.run_number }}/artifact/artifacts/${{ env.TARGET }}
+      - name: Update commit status
+        uses: ouzi-dev/commit-status-updater@v2
```
Comment (daltonbohning) [inline .github/workflows/rpm-build-and-test.yml:307]: Do we want to pin SHA for new additions?

**Review by daltonbohning — APPROVED** [review]

File: .github/workflows/rpm-build-and-test.yml (hunk 1 above)
Comment (brianjmurrell) [reply .github/workflows/rpm-build-and-test.yml:307]: Not IMO.  At least not for the sake of ticking a checkbox.  If we independently decide that that is a burden we want to take on for our own reasons, that is one thing.  But I think this blanket requirement to pin everything to a git hash is going to prove to be unworkable and get some push-back.  Or maybe we are an outlier with over a hundred of these dependencies that we'd have to manually track and update, etc.

Comment (daltonbohning) [reply .github/workflows/rpm-build-and-test.yml:307]: Like we discussed on #13865, we need more investigation on these things, so no problem from me.

**Review by ryon-jensen — APPROVED** [review]

File: .github/workflows/rpm-build-and-test.yml [hunk 2]
```diff
@@ -297,6 +303,12 @@ jobs:
...
+        with:
+          # yamllint disable-line rule:line-length
+          name: 'build/Build RPM on ${{ env.DISTRO_NAME }} ${{ env.COMMIT_STATUS_DISTRO_VERSION && env.COMMIT_STATUS_DISTRO_VERSION || env.DISTRO_VERSION }}'
```
Comment (daltonbohning) [inline .github/workflows/rpm-build-and-test.yml:310]: FYI, according to https://yaml-multiline.info/
Multi-line strings should work

**Comment by daltonbohning** [comment]
I don't see an immediate issue, but we should keep an eye out to make sure this continues to work as expected. This might also be one possibility for ftest-only, doc-only, etc.

**Comment by brianjmurrell** [comment]
> I don't see an immediate issue, but we should keep an eye out to make sure this continues to work as expected.

If it stops, it would be evident in a GHA run that doesn't set the _Required_ statuses to passing.

> This might also be one possibility for ftest-only, doc-only, etc.

Meaning to flip _Required_ commit statuses for the skipped stages?  If so, I have thought about that in the past.  It seems a bit dangerous though, as requiring gatekeepers to review the missing/incomplete _Required_ statuses seems like a good check and balance to prevent something that didn't actually do enough testing from slipping through.

**Comment by daltonbohning** [comment]
> > I don't see an immediate issue, but we should keep an eye out to make sure this continues to work as expected.
> 
> If it stops, it would be evident in a GHA run that doesn't set the _Required_ statuses to passing.

I'm more concerned with what you pointed out below: setting to "pass" when it didn't run or actually failed.

> > This might also be one possibility for ftest-only, doc-only, etc.
> 
> Meaning to flip _Required_ commit statuses for the skipped stages? If so, I have thought about that in the past. It seems a bit dangerous though, as requiring gatekeepers to review the missing/incomplete _Required_ statuses seems like a good check and balance to prevent something that didn't actually do enough testing from slipping through.

Agree. We would need to be very careful. But if done correctly, it would help prevent force landings. But I'm not close to tackling that problem yet :)

**Comment by brianjmurrell** [comment]
> Agree. We would need to be very careful. But if done correctly, it would help prevent force landings. But I'm not close to tackling that problem yet :)

I have often been of the mind that each PR should be able to have it's *Required*s configured for it specifically in some way.  Obviously a PR changing the rules/code for setting the *Required*s would need to be scrutinized for trying to evade them, or even better yet, perhaps it's the target branch's code that is used to set them so that a PR cannot change them for itself.

But ultimately, the static list as it is currently seems rather inflexible.

**Comment by daltonbohning** [comment]
> I have often been of the mind that each PR should be able to have it's _Required_s configured for it specifically in some way. Obviously a PR changing the rules/code for setting the _Required_s would need to be scrutinized for trying to evade them, or even better yet, perhaps it's the target branch's code that is used to set them so that a PR cannot change them for itself.
> 
> But ultimately, the static list as it is currently seems rather inflexible.

Another idea (not well thought out) could be to remove the requirement for each individual stage and add a requirement for some "overall" status/stage. Where that "overall" determines whether a PR complete the necessary stages. A benefit to this approach is that we wouldn't have to update the branch protection requirements after adding a new stage. The stage requirements would be version-controlled alongside the stage itself. The downside is the scrutiny as you mentioned

**Pull request merged**